File: LinkedListInsertEnd.java
class LinkedListInsertEnd{
    public static void main(String args[]){
        SingleLinkedList ll = new SingleLinkedList();

        ll.insertAtEnd(10);
        ll.insertAtEnd(20);
        ll.insertAtEnd(30);
        ll.insertAtEnd(40);

        ll.display();
    }
}

class sll{
    int data;
    sll next;

    sll(int data,sll next){
        this.data = data;
        this.next = next;
    }
}

class SingleLinkedList{
    sll newNode;
    sll head;
    SingleLinkedList(){
        this.head = null;
        this.newNode = null;
    }

    void createNode(int data){
        this.newNode = new sll(data,null);
    }

    void insertAtEnd(int data){
        if(this.head == null){
            this.createNode(data);
            this.head = this.newNode;
        }
        else{
            sll temp;
            temp = this.head;

            while(temp.next != null){
                temp = temp.next;
            }
            temp.next = this.newNode;
        }
    }

    void display(){
        sll temp = this.head;
        while(temp != null){
            System.out.println(temp.data);
            temp = temp.next;
        }
    }
}

File: LinkedList_insert_end.c
#include <stdio.h>
#include <stdlib.h>

typedef struct Node{
    int data;
    struct Node *next;
}sll;

sll* createNode(int data){
    sll *temp = (sll *)malloc(sizeof(sll));

    if(temp == NULL){
        printf("No Memory");
        exit(0);
    }

    temp->data = data;
    temp->next = NULL;

    return temp;
}

void insert_at_end(sll **node,int data){
    sll *newNode = createNode(data);

    if(*node == NULL){
        *node = newNode;
    }
    else {
        sll *temp = *node;
        
        while(temp->next != NULL){
            temp = temp->next;
        }
        temp->next = newNode;
    }
}

void display(sll *head){
    sll *temp = head;
    while(temp->next != NULL){
        printf("%d->",temp->data);
        temp = temp->next;
    }
    printf("%d",temp->data);
}

int main(void){
    sll *head = NULL;

    insert_at_end(&head,10);
    insert_at_end(&head,20);
    insert_at_end(&head,30);
    insert_at_end(&head,40);
    insert_at_end(&head,50);

    display(head);
    return 0;
}
